### PR TITLE
Fix single_reference=True

### LINKED ISSUE #42: Bug: single_reference=True not working currently
I think a bug has been introduced with the recent release - reproject makes use of the ``single_reference=True`` option but this doesn't appear to be working anymore in the last few days:

https://github.com/astropy/reproject/actions/runs/6854228148/job/18815976303

Example error:

```
E               Exception: File not found for comparison test
E                                               Generated file:
E                                               	/var/folders/3s/vfzpb5r51gs6y328rmlgzm7c0000gn/T/tmp8ftf61xs/test_reproject_roundtrip[asdf].fits
E                                               This is expected for new tests.
```

but the reference file should be ``test_reproject_roundtrip.fits`` not ``test_reproject_roundtrip[asdf].fits``

### PR DESCRIPTION
**Pull request by pllim**
Fix #42

### CODE REVIEW

**Comment by pllim** [comment]
Nope does not appear to work.

File: pytest_arraydiff/plugin.py [hunk 1]
```diff
@@ -310,7 +310,7 @@ def pytest_runtest_call(self, item):
         # Find test name to use as plot name
         filename = compare.kwargs.get('filename', None)
         if filename is None:
-            filename = item.name + '.' + extension
+            filename = item.function + '.' + extension
             if not single_reference:
                 filename = filename.replace('[', '_').replace(']', '_')
                 filename = filename.replace('_.' + extension, '.' + extension)
```
Comment (ConorMacBride) [inline pytest_arraydiff/plugin.py:316]: I think this should work:
```py
        if filename is None:
            if single_reference:
                filename = item.function + '.' + extension
            else:
                filename = item.name + '.' + extension
                filename = filename.replace('[', '_').replace(']', '_')
                filename = filename.replace('_.' + extension, '.' + extension)
```

Comment (pllim) [reply pytest_arraydiff/plugin.py]: Ah, thanks! I updated the patch and added you as co-author of the commit.

**Comment by pllim** [comment]
Yay, CI is green now. Thanks!

I have no expertise nor interest to add a unit test but @astrofrog is welcome to push to my branch if he wants a test. 😉

**Comment by astrofrog** [comment]
I've added a regression test, which unfortunately shows that the fix here doesn't work

File: pytest_arraydiff/plugin.py [hunk 2]
```diff
@@ -310,8 +310,10 @@ def pytest_runtest_call(self, item):
         # Find test name to use as plot name
         filename = compare.kwargs.get('filename', None)
         if filename is None:
-            filename = item.name + '.' + extension
-            if not single_reference:
+            if single_reference:
+                filename = item.function + '.' + extension
```
Comment (ConorMacBride) [inline pytest_arraydiff/plugin.py:314]: Whoops, it was accessing the name before https://github.com/astropy/pytest-arraydiff/commit/04e365cfbdc7992fbd2772107726706942aae374#diff-c5983910b8498a8d2d64db7dd4bf7a2cf6e7399a31a161fd2e5eed2f04484460L286
```suggestion
                filename = item.function.__name__ + '.' + extension
```

Comment (astrofrog) [reply pytest_arraydiff/plugin.py:314]: Let's try!

**Comment by astrofrog** [comment]
@ConorMacBride - for some reason the function name is ending up being ``wrapper`` so this doesn't work quite right.

**Comment by astrofrog** [comment]
Ok I got it to work, it was ``item.originalname`` we wanted - I'll go ahead and merge and release once CI passes, since this is needed for reproject.

**Comment by pllim** [comment]
Thanks, @astrofrog !

**Pull request merged**